### PR TITLE
FUSETOOLS2-2173 - Provide command to generate Camel route from OpenAPI

### PR DESCRIPTION
**Pull request by apupier**
https://github.com/camel-tooling/camel-lsp-client-vscode/assets/1105127/187083ba-b0d1-4ac8-b5ea-f6a6316a106e

### CODE REVIEW

File: src/requirements/CamelJBang.ts [hunk 1]
```diff
@@ -36,4 +36,15 @@ export class CamelJBang {
...
 		return new ShellExecution('jbang', [`'-Dcamel.jbang.version=${this.camelVersion}'`, 'camel@apache/camel', 'export', `--runtime=${runtime}`, `--gav=${gav}`]);
 	}
+
+	public generateRest(routefile: string, openApiFile: string) {
+		return new ShellExecution('jbang',
+			[`'-Dcamel.jbang.version=${this.camelVersion}'`,
+			'camel@apache/camel',
+			'generate',
+			'rest',
+			`--input=${openApiFile}`,
```
Comment (djelinek) [inline src/requirements/CamelJBang.ts:46]: I would consider to encapsulate also --input and --output arguments into single quotes

**Comment by djelinek** [comment]
1 code smell reported by SonarCloud

**Comment by apupier** [comment]
> I do not understand why but currently during checks is executed only one UI test

I surely left a `.only`

**Comment by apupier** [comment]
> 1 code smell reported by SonarCloud - https://sonarcloud.io/project/issues?id=camel-lsp-client-vscode&pullRequest=1600&resolved=false&types=CODE_SMELL

for the code smell, I prefer to not touch it in this PR. it is reported as new but in fact it is only moved code. I remember that I tried using the suggestion without success when the code was introduced.

File: src/ui-test/tests/commands.openapi.test.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import { WebDriver, SideBarView, VSBrowser, ActivityBar, EditorView, Workbench, InputBox, DefaultTreeSection } from "vscode-uitests-tooling";
+import { RESOURCES, waitUntilExtensionIsActivated, deleteFile, killTerminal, waitUntilEditorIsOpened } from "../utils/testUtils";
+import * as pjson from '../../../package.json';
+import { expect } from "chai";
+
+let driver: WebDriver;
+let input: InputBox;
+let sideBar: SideBarView;
+
+describe('Create a Camel Project using command', function () {
```
Comment (djelinek) [inline src/ui-test/tests/commands.openapi.test.ts:27]: label of this `describe` is not correct I would suggest to after change also modify the `it` label, eg. could be sufficient

```typescript
describe('Create a Camel Route using command', function () {
    it('OpenAPI', async function () {
```

**Comment by apupier** [comment]
> > > 1 code smell reported by SonarCloud - https://sonarcloud.io/project/issues?id=camel-lsp-client-vscode&pullRequest=1600&resolved=false&types=CODE_SMELL
> > 
> > 
> > for the code smell, I prefer to not touch it in this PR. it is reported as new but in fact it is only moved code. I remember that I tried using the suggestion without success when the code was introduced.
> 
> ok so maybe we can resolve directly in SonarCloud as "Won't fix"

I think we can keep it. If one day we have an expert of regexp in TypeScript, he/she can tackle it

**Pull request merged**